### PR TITLE
SCMOD-7700: Stop container if script execution fails

### LINKED ISSUE #13: Can Startup Scripts Halt The Startup Of The Image?
With the `/startup` approach where derived images can drop-in scripts that should be executed prior to an images `CMD` instruction being invoked, is there a way for the drop-in startup scripts to stop the image from starting?

I am looking for an invocation pattern example, where I can add checks in my startup scripts to indicate that an image should not start for things like:

* A database connection is not available after some short timeout period.
* The underlying database schema is not at the correct level.
* Required environment variables have not been set.

In my current `entrypoint.sh` script I perform these types of checks and stop the image with an error, but as I migrate to using the CAF startup facility I am looking for the correct approach to use.

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
Jira: https://portal.digitalsafe.net/browse/SCMOD-7700

### CODE REVIEW

File: src/main/docker/startup/startup.sh [hunk 1]
```diff
@@ -26,6 +26,13 @@ for script in $(dirname "$0")/startup.d/*; do
     if [ -x "$script" ]; then
         log "Running ${script##*/}..."
         "$script" |& $(dirname "$0")/../scripts/caf-log-format.sh "${script##*/}"
+        if [ ${PIPESTATUS[0]} -eq 0 ]
+        then
+            log "Successfully ran ${script##*/}"
```
Comment (dermot-hardy) [inline src/main/docker/startup/startup.sh]: There is no need for this extra unnecessary logging.

Comment (dermot-hardy) [reply src/main/docker/startup/startup.sh]: Get rid of the success branch.

**Comment by michael-bryson** [comment]
@dermot-hardy I know this is a bug fix but it still could be viewed as a breaking change, what do you think? It is changing the behaviour of how the image is starting up.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
That's a fair point.  It's possible that this _bug_ is being replied upon.  I agree we should treat this as a breaking change.  It will also be a breaking change in all downstream base images which move to it.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**